executor/executor.go

```diff
@@ -983,11 +983,12 @@ func newLockCtx(seVars *variable.SessionVars, lockWaitTime int64) *tikvstore.Loc
 		LockExpired:           &seVars.TxnCtx.LockExpire,
 		ResourceGroupTag:      resourcegrouptag.EncodeResourceGroupTag(sqlDigest, planDigest),
 		OnDeadlock: func(deadlock *tikverr.ErrDeadlock) {
-			// TODO: Support collecting retryable deadlocks according to the config.
-			if !deadlock.IsRetryable {
-				rec := deadlockhistory.ErrDeadlockToDeadlockRecord(deadlock)
-				deadlockhistory.GlobalDeadlockHistory.Push(rec)
+			cfg := config.GetGlobalConfig()
+			if deadlock.IsRetryable && !cfg.PessimisticTxn.DeadlockHistoryCollectRetryable {
```

Comment:
If a retryable deadlock occurs many times, can it pollute the deadlock histroy?

Reply:
Yes. So I make it not collected by default.

Reply:
I'm afraid it's never turned on because it can't be changed online. For retryable deadlock, it's better to merge the error from the same statements and record its retry count.

Reply:
I plan to make it support online-changing via the http api. But,  you are right, merging repeated error is a good idea. I'll reconsider it.

Reply:
To make an record to become latest when updating the retry count of an existing record, It seems we need to change the deadlock history collection (which is a simple queue now) into an LRU. @youjiali1995

Reply:
I didn't support merging repeated error for now since it needs too much code change, but I added the HTTP API. PTAL.